File: 45_SortArrayForMinNumber/SortArrayForMinNumber.py
class Solution:
    def PrintMinNumber(self, numbers):
        # write code here
        if not numbers:
            return ''
        tmp = sorted(
            numbers,
            cmp=lambda x, y: int(str(x) + str(y)) - int(str(y) + str(x)))
        return ''.join([str(x) for x in tmp])

File: 45_SortArrayForMinNumber/SortArrayForMinNumber.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
using namespace std;

class Solution
{
public:
    string PrintMinNumber(vector<int> numbers)
    {
        if (numbers.size() == 0)
            return "";
        sort(numbers.begin(), numbers.end(), cmp);
        string res;
        for (int i = 0; i < numbers.size(); ++i)
            res += to_string(numbers[i]);
        return res;
    }

    static bool cmp(int a, int b)
    {
        string A = to_string(a) + to_string(b);
        string B = to_string(b) + to_string(a);
        return A < B;
    }
};